### DISCUSSION TITLE
[Feature request] Local embedded videos

### DISCUSSION THREAD

**Discussion by seanpm2001** [community/community#124020] [Repositories]
### Select Topic Area

Product Feedback

### Body

***

# [Feature request] Local embedded videos

Right now, when you put a video into a repository, it has to work solely through GitHub online storage (github-usercontent) and doesn't work when you try to link a local repository video. It would be great if the option to embed videos into Markdown could be done without relying on this server. This helps with preservation, and cross-platform support (when you import the repository to another site)

***

**Comment by Rishit-katiyar** [comment]
Hi,

1. **Convert Video to a Web-Compatible Format**: Ensure your video file is in a format supported by web browsers, such as MP4, WebM, or Ogg. You can use tools like ffmpeg to convert your video if needed:

   ```bash
   ffmpeg -i input_video.avi -c:v libx264 -crf 23 -c:a aac -strict experimental -b:a 128k output_video.mp4
   ```

2. **Upload Video to Your Repository**: Place your converted video file in your GitHub repository. Make sure it's in a directory that's accessible to your Markdown file.

3. **Embed Video in Markdown**: In your Markdown file, use the HTML `<video>` tag to embed the video. Provide the path to your video file relative to your Markdown file:

   ```markdown
   <video width="320" height="240" controls>
     <source src="path/to/your/video.mp4" type="video/mp4">
     Your browser does not support the video tag.
   </video>
   ```

   Replace `"path/to/your/video.mp4"` with the actual path to your video file within your repository.

4. **Commit and Push Changes**: After embedding the video in your Markdown file, commit the changes to your repository and push them to GitHub:

   ```bash
   git add your_markdown_file.md
   git commit -m "Added embedded video"
   git push origin main
   ```

5. **Verify on GitHub**: Visit your repository on GitHub and navigate to the Markdown file where you embedded the video. You should now see the video embedded in the file.
